player/screenshot.c

```diff
@@ -495,12 +514,14 @@ void cmd_screenshot_to_file(void *p)
     if (!image) {
         mp_cmd_msg(cmd, MSGL_ERR, "Taking screenshot failed.");
         cmd->success = false;
-        return;
+        goto done;
     }
     char *path = mp_get_user_path(NULL, mpctx->global, filename);
     cmd->success = write_screenshot(cmd, image, path, &opts, true);
     talloc_free(image);
     talloc_free(path);
```

Comment:
You could use `tmp` for the talloc_ctx for the user path here.

Reply:
done